src/Presentation/Controllers/ExpenseController.cs

```diff
@@ -0,0 +1,115 @@
+using ExpenseTrackerGrupo4.src.Aplication.Interfaces;
+using ExpenseTrackerGrupo4.src.Presentation.DTOs;
+using AutoMapper;
+using ExpenseTrackerGrupo4.src.Domain.Entities;
+
+namespace ExpenseTrackerGrupo4.src.Presentation.Controllers;
+
+[ApiController]
+[Route("api/expenses")]
+[Authorize]
+public class ExpenseController(IExpenseService expenseService, IMapper mapper) : ControllerBase
+{
+    private readonly IExpenseService _expenseService = expenseService;
+    private readonly IMapper _mapper = mapper;
+
+    [HttpPost]
+    [Authorize]
+    public async Task<IActionResult> CreateExpense([FromBody] CreateUpdateExpenseDto dto)
+    {
+        var userId = GetCurrentUserId();
+
+        if(userId == Guid.Empty) return Forbid(); 
+
+        var expense = _mapper.Map<Expense>(dto);
+        expense.UserId = userId; 
+
+        await _expenseService.AddAsync(expense);
+        return CreatedAtAction(nameof(GetExpenseById), new { id = expense.Id }, dto);
+    }
+
+    [HttpGet]
+    [Authorize]
+    public async Task<IActionResult> GetUserExpenses(
+        [FromQuery] DateTime? startDate, [FromQuery] DateTime? endDate, [FromQuery] string? category
+    )
+    {
+        var userId = GetCurrentUserId();
+
+        if(userId == Guid.Empty) return Forbid(); 
+
+        var expenses = await _expenseService.GetUserExpensesCommand(userId, startDate, endDate, category);
+
+        return Ok(expenses);
+    }
+
+    [HttpGet("{id}")]
+    [Authorize]
+    public async Task<IActionResult> GetExpenseById(Guid id)
+    {
+        var userId = GetCurrentUserId();
+        try
+        {
+            var expense = await _expenseService.GetByIdAsync(id, userId);
+            if (expense == null) return NotFound();
+            return Ok(expense);
+        }
+        catch (UnauthorizedAccessException)
+        {
+            return Forbid(); 
+        }
+
+    }
+
+    [HttpPut("{id}")]
+    [Authorize]
+    public async Task<IActionResult> UpdateExpense(Guid id, [FromBody] CreateUpdateExpenseDto dto)
+    {
+        var userId = GetCurrentUserId();
+        var existingExpense = await _expenseService.GetByIdAsync(id, userId);
+
+        if (existingExpense == null) return NotFound();
+
+        var expenseToUpdate = _mapper.Map<Expense>(dto);
+        expenseToUpdate.UserId = userId; 
+
+        try
+        {
+            await _expenseService.UpdateAsync(expenseToUpdate, userId);
+            return NoContent();
+        }
+        catch (UnauthorizedAccessException)
+        {
+            return Forbid(); 
+        }
+    }
+
+    [HttpDelete("{id}")]
+    [Authorize]
+    public async Task<IActionResult> DeleteExpense(Guid id)
+    {
+        var userId = GetCurrentUserId();
+        var existingExpense = await _expenseService.GetByIdAsync(id, userId);
+
+        if (existingExpense == null) return NotFound();
+
+        try
+        {
+            await _expenseService.DeleteAsync(id, userId);
+            return NoContent(); 

```

Comment:
An ok could be returned instead of a NoContent to avoid having a 204 response that is undocumented or document the 204 response.